examples/src/main/java/com/datastax/oss/driver/examples/json/jackson/JacksonJsonFunction.java

```diff
@@ -0,0 +1,229 @@
+ *       identified by basic.contact-points (see application.conf).
+ * </ul>
+ *
+ * <p>Side effects:
+ *
+ * <ul>
+ *   <li>creates a new keyspace "examples" in the cluster. If a keyspace with this name already
+ *       exists, it will be reused;
+ *   <li>creates a user-defined type (UDT) "examples.json_jackson_function_user". If it already
+ *       exists, it will be reused;
+ *   <li>creates a table "examples.json_jackson_function". If it already exists, it will be reused;
+ *   <li>inserts data in the table.
+ * </ul>
+ *
+ * @see <a href="http://www.datastax.com/dev/blog/whats-new-in-cassandra-2-2-json-support">What’s
+ *     New in Cassandra 2.2: JSON Support</a>
+ */
+public class JacksonJsonFunction {
+
+  // A codec to convert JSON payloads into User instances;
+  private static final TypeCodec<User> USER_CODEC = new JacksonJsonCodec<>(User.class);
+
+  // A codec to convert generic JSON payloads into JsonNode instances
+  private static final TypeCodec<JsonNode> JSON_NODE_CODEC = new JacksonJsonCodec<>(JsonNode.class);
+
+  public static void main(String[] args) {
+    try (CqlSession session =
+        new CqlSessionBuilder().addTypeCodecs(USER_CODEC, JSON_NODE_CODEC).build()) {
+      createSchema(session);
+      insertFromJson(session);
+      selectToJson(session);
+    }
+  }
+
+  private static void createSchema(CqlSession session) {
+    session.execute(
+        "CREATE KEYSPACE IF NOT EXISTS examples "
+            + "WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1}");
+    session.execute(
+        "CREATE TYPE IF NOT EXISTS examples.json_jackson_function_user(" + "name text, age int)");
+    session.execute(
+        "CREATE TABLE IF NOT EXISTS examples.json_jackson_function("
+            + "id int PRIMARY KEY, user frozen<json_jackson_function_user>, scores map<varchar,float>)");
+  }
+
+  // Mapping JSON payloads to table columns of arbitrary types,
+  // using fromJson() function
+  private static void insertFromJson(CqlSession session) {
+
+    User alice = new User("alice", 30);
+    User bob = new User("bob", 35);
+
+    ObjectNode aliceScores =
+        JsonNodeFactory.instance.objectNode().put("call_of_duty", 4.8).put("pokemon_go", 9.7);
+    ObjectNode bobScores =
+        JsonNodeFactory.instance.objectNode().put("zelda", 8.3).put("pokemon_go", 12.4);
+
+    // Build and execute a simple statement
+    Statement stmt =
+        insertInto("examples", "json_jackson_function")
+            .value("id", literal(1))
+            // client-side, the User object will be converted into a JSON String;
+            // then, server-side, the fromJson() function will convert that JSON string
+            // into an instance of the json_jackson_function_user user-defined type (UDT),
+            // which will be persisted into the column "user"
+            .value("user", function(CqlIdentifier.fromCql("fromJson"), literal(alice, USER_CODEC)))
```

Comment:
We should use the string variant of `function` for simplicity:
```suggestion
            .value("user", function("fromJson", literal(alice, USER_CODEC)))
```
The examples already do for everything else (table name, bind marker, etc).